### PR TITLE
Apply README conventions.

### PR DESCRIPTION
**Pull request by jrobinAV**
I applied a similar README convention on all the repos.

Please fill the holes I was not able to fill. There are several remaining TODOs, particularly in INSTALLATION.md and Directory structure.

#90 Improve Gui Doc

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
+# Contributions
+
+Thanks for your interest in helping improve Taipy! Contributions are welcome, and they are greatly appreciated!
+Every little help and credit will always be given.
```
Comment (FredLL-Avaiga) [inline CONTRIBUTING.md]: Every little helps, credit will always be given.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+There are multiple ways to contribute to Taipy, code, but also reporting bugs, creating feature requests, helping
```
Comment (FredLL-Avaiga) [inline CONTRIBUTING.md]: doc ?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+other users in our forums, Stack Overflow, etc.
```
Comment (FredLL-Avaiga) [inline CONTRIBUTING.md]: StackOverflow in 1 word

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+## Never contributed on an open source project before ?
+
+Have a look on this [GitHub documentation](https://docs.github.com/en/get-started/quickstart/contributing-to-projects).
+
+## Report bugs
+
+Reporting bugs is through [GitHub issues](https://github.com/Avaiga/taipy/issues).
+
+Please report relevant information and preferably code that exhibits the problem. We provide templates to help you
```
Comment (FredLL-Avaiga) [inline CONTRIBUTING.md:19]: to help you precisely describe the issue

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+## Coding style and best practices
+
+### Python
+
+Taipy's repositories follow the [PEP 8](https://www.python.org/dev/peps/pep-0008/) and
+[PEP 484](https://www.python.org/dev/peps/pep-0484/) coding convention.
+
+## Javascript
+
+Taipy's repositories follow the [W3Schools](https://www.w3schools.com/js/js_conventions.asp) and
```
Comment (FredLL-Avaiga) [inline CONTRIBUTING.md]: We have almost no js code it's all Typescript and we use the Prettier VS code plugin ...

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+### Git branches
+
+All new development happens in the `develop` branch. All pull requests should target that branch.
+We are following a strict branch naming convention based on the pattern: `<type>/#<issueId>[IssueSummary]`.
+
+Where:
+
+- `<type>` would be one of:
+    - feature: new feature implementation, or improvement of a feature.
+    - fix: bug fix.
```
Comment (FredLL-Avaiga) [inline CONTRIBUTING.md]: I thought this was bug ?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+2. We are working with [Pipenv](https://github.com/pypa/pipenv) for our virtualenv.
+   Create a local env and install development package by running `pipenv install --dev`, then run tests with `pipenv
+   run pytest` to verify your setup.
+
+3. For convention help, we provide a [pre-commit](https://pre-commit.com/hooks.html) file.
+   This tool will run before each commit and will automatically reformat code or raise warnings and errors based on the
+   code format or Python typing.
+   You can install and setup it up by doing:
+   ```
+     pipenv install pre-commit --skip-lock
```
Comment (FredLL-Avaiga) [inline CONTRIBUTING.md]: no need for --skip-lock I guess

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+## Dependency management
+
+Taipy comes with multiple optional packages. You can find the list directly in the product or Taipy's packages.
+The back-end Pipfile does not install by default optional packages due to `pyodbc` requiring a driver's manual
+installation. This is not the behaviour for the front-end that installs all optional packages through its Pipfile.
+
+If you are a contributor on Taipy, be careful with dependencies, do not forget to install or uninstall depending on
+your issue.
+
+If you need to add to Taipy a new dependency, do not forget to add it in the `Pipfile` and the `setup.py`.
```
Comment (FredLL-Avaiga) [inline CONTRIBUTING.md]: to Taipy a new dependency => a new dependency to Taipy

File: README.md [hunk 9]
```diff
@@ -1,45 +1,64 @@
...
+- [License](#license)
+- [Usage](#usage)
+- [Taipy GUI](#what-is-taipy-gui)
+- [Installation](#installation)
+- [Contributing](#contributing)
+- [Code of conduct](#code-of-conduct)
+- [Directory Structure](#directory-structure)
 
-!!! abstract "TODO: fix PYPI package or version link"
+## What is Taipy GUI
```
Comment (FredLL-Avaiga) [inline README.md:23]: is it Taipy GUI or Taipy Gui or Taipy-gui ?

File: README.md [hunk 10]
```diff
@@ -1,45 +1,64 @@
...
 
-The files needed to build _Taipy_ are located in sub-directories that
-store files as follows:
-
--   `taipy`: Python source files
--   `gui`: Graphical user interface related files
--   `docs`: The current documentation
-
-!!! abstract "TODO: complete the directory structure"
+- `demos`: TODO or to remove.
```
Comment (FredLL-Avaiga) [inline README.md]: I'd say remove

**Review by FredLL-Avaiga — APPROVED** [review]
a few comments

**Review by dinhlongviolin1 — APPROVED** [review]

File: README.md (hunk 9 above)
Comment (jrobinAV) [reply README.md:23]: In the title, its Taipy GUI. other wise the repo name is taipy-gui. I guess it should never be Taipy Gui

File: README.md [hunk 11]
```diff
@@ -1,45 +1,62 @@
...
+- `CONTRIBUTING.md`: Instructions to contribute to _taipy-gui_;
+- `INSTALLATION.md`: Instructions to install _taipy-gui_;
+- `LICENSE`: The Apache 2.0 License;
+- `MANIFEST.in`: Build configuration file;
+- `mypy.ini`: [mypy](http://mypy-lang.org/) linter configuration file;
+- `Pipfile`: File used by the Pipenv virtual environment to manage project dependencies;
+- `pyproject.toml`: Python build configuration file use for linters here;
+- `pytest.ini`: [pytest](https://pytest.org/) configuration file;
+- `README.md`: Current file;
+- `setup.py`: The setup script managing building, distributing, and installing _taipy-rest_;
```
Comment (dinhlongviolin1) [inline README.md]: replace taipy-rest with taipy-gui

**Review by dinhlongviolin1 — APPROVED** [review]

**Review by FabienLelaquais — APPROVED** [review]
Excellent

File: README.md (hunk 11 above)
Comment (FredLL-Avaiga) [reply README.md]: good catch

**Pull request merged**